bam/msa configuration

rohinikumart opened the pull request:
##### SUMMARY
Configuration for BAM/MSA on interface.

##### GitHub Issues
List the GitHub issues impacted by this PR. If no Github issues are affected, please indicate this with "N/A".

| GitHub Issue # |
| -------------- |
| |
N/A

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request
- Docs Pull Request
- Feature Pull Request
- Test Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->

##### OUTPUT
<!--- Paste the functionality test result below -->
```paste below

```
##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```
<!--- Measure the code coverage before and after the change by running the UT and ensure that the "coverage after the change" is not less than the coverage "before the change". Note that the unit testing coverage can be manually executed using the pytest tool or ansible-test tool. -->

##### Checklist:

- [x] I have performed a self-review of my own code to ensure there are no formatting, linting, or security issues
- [x] I have verified that new and existing unit tests pass locally with my changes
- [x] I have not allowed coverage numbers to degenerate
- [x] I have maintained at least 90% code coverage
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have maintained backward compatibility or have provided any relevant "breaking_changes" descriptions in a "fragment" file in the "changelogs/fragments" directory of this repository.
- [x] I have provided a summary for this PR in valid "fragment" file format in the "changelogs/fragments" directory of this repository branch. Reference : [Ansible Change Log Document](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)

##### How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration
default fec is RS in z9864. while running regression, changed fec from FEC_DISABLED to FEC_RS in plugins/module_utils/network/sonic/config/interfaces/interfaces.py
- [ ] Test A
- [ ] Test B

test results : 
[regression-2025-09-03-16-34-08.html.pdf](https://github.com/user-attachments/files/22116567/regression-2025-09-03-16-34-08.html.pdf)

kerry-meyer commented:
Thank you for re-synching and resolving the conflicts.

The sanity and coverage tests have been run after your update.

There is one sanity error that needs to be fixed:

`ERROR: Found 1 pep8 issue(s) which need to be resolved:
ERROR: plugins/module_utils/network/sonic/argspec/interfaces/interfaces.py:77:48: E127: continuation line over-indented for visual indent`

Also, the UT coverage for the sonic_interfaces module has dropped a little bit. Please add a UT deletion case for AUTONEG_MODE_MSA to at least partially offset this reduction.

There are also a couple of minor pending comments for the "module" file (plugins/modules/sonic_interfaces.py).

rohinikumart commented on plugins/modules/sonic_interfaces.py:
I have updated example. Please check .

rohinikumart replied:
Added them now separately

rohinikumart commented:
> Thank you for re-synching and resolving the conflicts.
> 
> The sanity and coverage tests have been run after your update.
> 
> There is one sanity error that needs to be fixed for line 77 in the ''argspec" file:
> 
> `ERROR: Found 1 pep8 issue(s) which need to be resolved: ERROR: plugins/module_utils/network/sonic/argspec/interfaces/interfaces.py:77:48: E127: continuation line over-indented for visual indent`
> 
> Also, the UT coverage for the sonic_interfaces module has dropped a little bit. Please add a UT deletion case for AUTONEG_MODE_MSA to at least partially offset this reduction.
> 
> There are also a couple of minor pending comments for the "module" file (plugins/modules/sonic_interfaces.py).

added delete for AUTONEG_MODE_MSA and addressed other comments

rohinikumart commented on tests/unit/modules/network/sonic/fixtures/sonic_interfaces.yaml:
added in replaced_01 and overridden_01

rohinikumart commented on tests/regression/roles/sonic_interfaces/defaults/main.yml:
```diff
@@ -68,6 +76,8 @@ sonic_interfaces_tests:
       - name: "{{ interface7 }}"
         description:
+        # Use only on platform that support this.
+        #autoneg_mode: AUTONEG_MODE_MSA
```
default is BAM. when we delete , it has to go to default

kerry-meyer commented on plugins/modules/sonic_interfaces.py:
```diff
@@ -97,6 +98,15 @@
         - UNRELIABLE_LOS_MODE_AUTO
+      autoneg_mode:
+        description:
+        - BAM/MSA configuration for autonegotiation
+        - Applicable only for Ethernet interfaces.
```
Please clarify the default behavior for autoneg_mode.

For example:

```suggestion
        - BAM/MSA configuration for autonegotiation
        - Applicable only for Ethernet interfaces.
        - AUTONEG_MODE_BAM is set by default if "auto_negotiate" is set to "true".
```
